Feature/dx 638 3ds2 (#518)
SwedbankPay/developer.swedbankpay.com: payments/card/other-features.md

{% include prices.md %}

## 3D Secure 2.0

When dealing with card payments, 3D Secure authentication of the
cardholder is an essential topic. 3D Secure 2.0 is an improved version of the
old protocol, now allowing frictionless payments where transactions can be
completed without input from the cardholder. Therefore, there are certain fields

> that should be included when implementing 3D Secure 2.0. These are listed below: 

Reviewer: ```suggestion
## 3D Secure 2

When dealing with card payments, 3-D Secure authentication of the cardholder is
an essential topic. 3-D Secure 2.0 is an improved version of the 3-D Secure
protocol, now allowing frictionless payments where transactions can be completed
without input from the cardholder. Therefore, there are certain fields that
should be included to allow frictionless 3-D Secure 2.0. These fields are listed
below: 

```

> *  `payment.riskIndicator`

Reviewer 2: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5786122)

> *  `payment.riskIndicator`

Reviewer 2: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5786122)

> *  `payment.cardholder`

Reviewer 2: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5786122)

> *  `payment.cardholder`

Reviewer 2: Codacy found an issue: [[list-item-indent] Incorrect list-item indent: add 1 space](https://app.codacy.com/gh/SwedbankPay/developer.swedbankpay.com/pullRequest?prid=5786122)

> *   `payment.cardholder`

Reviewer: Perhaps we should enlist all fields within `riskIndicator` and `cardHolder` by just providing a full JSON code example?

{% include prices.md %}

> ## 3-D Secure 2.0

Reviewer: ```suggestion
## 3-D Secure 2
```

When dealing with card payments, 3-D Secure authentication of the
cardholder is an essential topic. 3-D Secure 2.0 is an improved version of the
old protocol, now allowing frictionless payments where transactions can be
completed without input from the cardholder. Therefore, there are certain fields
that should be included when implementing 3-D Secure 2.0. These are listed below
and can be seen in the abbreviated request example below.

*   `payment.riskIndicator`

> *   `payment.cardholder`

Reviewer: This list can be removed when we have a full example, but let's add a field table below the example that highlights the 3-D Secure 2 related fields.

        },
        "riskIndicator": {
            "deliveryEmailAddress": "olivia.nyhuus@payex.com",
            "deliveryTimeFrameIndicator": "01",
            "preOrderDate": "19801231",
            "preOrderPurchaseIndicator": "01",
            "shipIndicator": "01",
            "giftCardPurchase": false,
            "reOrderPurchaseIndicator": "01",
        }
    }
}
```

{:.table .table-striped}
| Required          | Field                          | Data type | Description                                                                                                                                                      |
| :---------------- | :----------------------------- | :-------- | :--------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `payment.cardholder` | `object`  | Cardholder object that can hold information about a buyer (private or company). The information added increases the chance for [frictionless 3-D Secure 2 flow][3ds2].                                                                                                                      |

> | `payment.riskIndicator` | `object`  | This object consist of information that helps verifying the payer. Providing these fields decreases the likelihood of having to promt for [3-D Secure authentication][3ds2] of the payer when they are authenticating the purchase.                                                                                                                                                                                       |

Reviewer: Please enumerate all fields, not just the top-level ones.

{% include prices.md %}

## 3-D Secure 2

When dealing with card payments, 3-D Secure authentication of the
cardholder is an essential topic. 3-D Secure 2 is an improved version of the
old protocol, now allowing frictionless payments where transactions can be
completed without input from the cardholder. Therefore, there are certain fields
that should be included when implementing 3-D Secure 2. These are listed below
and can be seen in the abbreviated request example below.

{:.code-header}
**Request**

```http
POST /psp/creditcard/payments HTTP/1.1
Authorization: Bearer <AccessToken>
Content-Type: application/json

{
    "payment": {
        "operation": "Purchase",
        "intent": "Authorization",
        "currency": "SEK",
        "description": "Test Purchase",
        "urls": {
            "hostUrls": ["https://example.com"]
        },
        "cardholder": {
            "firstName": "Olivia",
            "lastName": "Nyhuus",
            "email": "olivia.nyhuus@payex.com",
            "msisdn": "+4798765432",
            "homePhoneNumber": "+4787654321",
            "workPhoneNumber": "+4776543210",
            "shippingAddress": {
                "firstName": "Olivia",

>                 "lastName": "Nyhuus",

Reviewer: Should it be `firstName` and `lastName` or `addressee`?

| └➔&nbsp;`workPhoneNumber`             | `string` | Payer's registered work phone number. The field is related to [3-D Secure 2][3ds2].                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |
| └─➔&nbsp;`shippingAddress`            | `object` | The shipping address object related to the `payer`. The field is related to [3-D Secure 2][3ds2].                                                                                                                                                                                                                                                                                                                                                                                                                                                                |

> | └─➔&nbsp;`addresse`                   | `object` | The shipping address object related to the `payer`.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |

Reviewer: I can't find `addressee` in the code example?

| └─➔&nbsp;`city`                       | `string` | Payer's city of residence                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
| └─➔&nbsp;`countryCode`                | `string` | Country Code for country of residence.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |

> | └➔&nbsp;`riskIndicator`               | `string` | This object consist of information that helps verifying the payer. Providing these fields decreases the likelihood of having to promt for [3-D Secure authentication][3ds2] of the payer when they are authenticating the purchase.                                                                                                                                                                                                                                                                                                                              |

Reviewer: ```suggestion
| └➔&nbsp;`riskIndicator`               | `object` | This object consist of information that helps verifying the payer. Providing these fields decreases the likelihood of having to promt for [3-D Secure authentication][3ds2] of the payer when they are authenticating the purchase.                                                                                                                                                                                                                                                                                                                              |
```

{% include iterator.html prev_href="after-payment" prev_title="Back: After
payment"  %}

> 

Reviewer: ```suggestion
```
